### PR TITLE
Add "cut point space" property

### LINKED ISSUE #939: Property Suggestion: Has a non-cut point (or Cut point space)
## Property Suggestion

A point $p \in X$ is called a *noncut point* if $X$ is connected and $p$ is not a [cut point](https://topology.pi-base.org/properties/P000204). I.e., $X$ and $X \backslash \\{p\\}$ are connected.

$X$ is a *Cut point space* if $X$ is nonempty, connected, and does not have a noncut point. Equivalently, $X$ is nonempty, connected, and every point is a cut point.

## Rationale

Engelking and Honari-Bahrampour, the references at  [Has a cut point (P204)](https://topology.pi-base.org/properties/P000204), define noncut point. Honari-Bahrampour defines cut point space. There is a [wikipedia page for cut point](https://en.wikipedia.org/wiki/Cut_point). danflapjax already has [a branch](https://github.com/pi-base/data/compare/main...cut-point) introducing this. 

## Relationship to other properties
It seems reasonable to consider adding either 'Has a noncut point' or 'Cut point space', but I'll just stick to 'Cut point space' here.

1. 'Cut point space => Has a cut point'. (Compare: '~Has cut point + Connected + ~Empty => Has a noncut point')

(Definition) 

2. 'Cut point space => ~Compact'. (Compare: Connected  + ~Empty + Compact => Has a noncut point)

 > [Honari-Bahrampour,  Corollary 3.10.] Let X be a cut-point space. Then X is non-compact.

3. 'Homogeneous + Has a cut point => Cut point space'. (Compare: 'Has a cut point + Has a noncut point => ~Homogeneous')

4. 'Cut point space => Has a closed point'. (Compare: Connected + ~Has a closed point => Has a noncut point)

> [Honari-Bahrampour, Theorem 3.7.] Let X be a cut-point space. Then the set of closed points of X is infinite.

This is related to 'Has a cut point + ~Has an isolated point => Has a closed point'.

## Remarks

- Engelking uses "noncut point". Honari-Bahrampour uses "non-cut point". Both non-cut point and noncut point seem common enough. (As are cut point space and cut-point space.)
- You can see on [Google scholar](https://scholar.google.com/scholar?start=0&q=%22cut+point+space%22&hl=en&as_sdt=0,21) that cut point space, cut-point space, cutpoint space all appear. There are 30 results.
- ["non-cut point"](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C21&q=%22non-cut+point%22&btnG=) has 294 results.
- The 'Has a cut point' property was introduced here https://github.com/pi-base/data/pull/936#issue-2681381592.
- 'Has a non-cut point' mentioned here https://github.com/pi-base/data/pull/302#issuecomment-2492395637

### PR DESCRIPTION
**Pull request by danflapjax**
Resolves #939

### CODE REVIEW

**Comment by david20000813** [comment]
P204 already implies P36 (T452), so T635 is not needed.

File: theorems/T000638.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+uid: T000638
+if:
+  P000205: true
+then:
+  P000001: true
+---
+Removing one of a pair of topologically indistinguishable points can't change connectivity, so a {P193} must be {P1}.
```
Comment (david20000813) [inline theorems/T000638.md:8]:
```suggestion
Removing one of a pair of topologically indistinguishable points can't change connectivity, so a {P204} must be {P1}.
```

File: theorems/T000639.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+uid: T000639
+if:
+  P000205: true
+then:
+  P000078: false
+refs:
+- doi: 10.1090/s0002-9939-99-04839-x
+  name: Cut-Point Spaces
```
Comment (prabau) [inline theorems/T000639.md:9]: Add the names of the authors in parentheses (same for other traits)

File: theorems/T000641.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+uid: T000641
+if:
+  P000205: true
+then:
+  P000107: true
+refs:
+- doi: 10.1090/s0002-9939-99-04839-x
+  name: Cut-Point Spaces
+---
+Theorem 3.7 in {{doi:10.1090/s0002-9939-99-04839-x}} states that a cut point space has infinitely many closed points.
```
Comment (prabau) [inline theorems/T000641.md:11]: Saying "states that" seems that it is just stated, but not necessily shown/proved.
How about saying instead: "By theorem ..., a cut point space ..."

**Comment by david20000813** [comment]
Non-compact (P16) already implies non-finite (P78), so T639 is redundant - it follows from T640.

File: theorems/T000640.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+uid: T000640
+if:
+  P000205: true
+then:
+  P000016: false
+refs:
+- doi: 10.1090/s0002-9939-99-04839-x
+  name: Cut-Point Spaces
```
Comment (david20000813) [inline theorems/T000640.md:9]:
```suggestion
  name: Cut-Point Spaces (Honari & Bahrampour)
```

File: theorems/T000641.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+uid: T000641
+if:
+  P000205: true
+then:
+  P000107: true
+refs:
+- doi: 10.1090/s0002-9939-99-04839-x
+  name: Cut-Point Spaces
```
Comment (david20000813) [inline theorems/T000641.md:9]:
```suggestion
  name: Cut-Point spaces (Honari & Bahrampour)
```

File: theorems/T000637.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+---
+uid: T000637
+if:
+  P000205: true
+then:
+  P000001: true
+---
+Removing one of a pair of topologically indistinguishable points can't change connectivity, so a {P204} must be {P1}.
```
Comment (david20000813) [inline theorems/T000637.md]:
```suggestion
Removing one of a pair of topologically indistinguishable points can't change connectivity, so a {P205} must be {P1}.
```

Sorry, made a typo in my last suggestion

**Comment by david20000813** [comment]
I guess one final question: should we reference P204 and P205 in each other's description? They both define "cut points", after all.

**Comment by david20000813** [comment]
I think this is ready now. But I'll leave it to @danflapjax to check and @prabau to do a final review, if you wish?

**Comment by prabau** [comment]
Looks good.  Approving and leaving it to @danflapjax to merge.

**Review by prabau — APPROVED** [review]

**Pull request merged**